firebase-config/src/main/java/com/google/firebase/remoteconfig/FirebaseRemoteConfig.java

```diff
@@ -652,6 +652,25 @@ private Task<Void> setDefaultsWithStringsMapAsync(Map<String, String> defaultsSt
         FirebaseExecutors.directExecutor(), (unusedContainer) -> Tasks.forResult(null));
   }
 
+  /**
+   * Asynchronously changes the custom signals for this {@link FirebaseRemoteConfig} instance.
+   *
+   * <p>The {@code customSignals} parameter should be an instance of {@link CustomSignals}, which
+   * enforces the allowed types for custom signal values (String, Long or Double).
+   *
+   * @param customSignalsMap A dictionary of keys and the values of the custom signals to be set for
+   *                         the app instance
+   */
+  @NonNull
+  public Task<Void> setCustomSignals(@NonNull CustomSignals customSignalsMap) {
+    return Tasks.call(
+        executor,
+        () -> {
+          frcSharedPrefs.setCustomSignals(customSignalsMap.customSignals);
```

Comment:
It's a little confusing, your `customSignals` here is actually the map. Why not just have `CustomSignals customSignals` in L665 and do something like `customSignals.getSignals()` here

Reply:
To maintain consistency with setCustomKeys in Crashlytics (and since we're not adding a get method), I've renamed customSignalsMap to customSignals.
https://github.com/firebase/firebase-android-sdk/blob/main/firebase-crashlytics/src/main/java/com/google/firebase/crashlytics/FirebaseCrashlytics.java#L395